components/chainhook-sdk/src/observer/http.rs

```diff
@@ -67,12 +67,33 @@ pub async fn handle_new_bitcoin_block(
                         e.to_string()
                     )
                 });
-                return Json(json!({
-                    "status": 500,
-                    "result": "unable to retrieve_full_block",
-                }));
+                None
             }
         };
+    let Some(block) = block else {
+        ctx.try_log(|logger| {
+            slog::crit!(
+                logger,
+                "Could not download bitcoin block after receiving new_burn_block. Exiting Chainhook observer."
+            )
+        });
+        match background_job_tx.lock() {
+            Ok(tx) => {
+                let _ = tx.send(ObserverCommand::Terminate);
+            }
+            Err(e) => {
+                ctx.try_log(|logger| {
+                    slog::crit!(logger, "Could not shut down event observer: {e}")
+                });
+                panic!("Could not shut down event observer: {e}")
```

Comment:
should we `std::process::exit(1)` instead, just to be 100% certain that we're terminating the process?

Reply:
Done!